File: samples/Scala/count-the-coins.scala
def countChange(amount: Int, coins:List[Int]) = {
	  val ways = Array.fill(amount + 1)(0)
	  ways(0) = 1
	  coins.foreach (coin =>
	  for (j<-coin to amount)
		  ways(j) =  ways(j) + ways(j - coin)
		  )
	ways(amount)
  }

countChange (15, List(1, 5, 10, 25))

File: samples/Mathematica/count-the-coins.m
CountCoins[amount_, coinlist_] := ( ways = ConstantArray[1, amount];
Do[For[j = coin, j <= amount, j++,
  If[ j - coin == 0,
    ways[[j]] ++,
    ways[[j]] += ways[[j - coin]]
]]
, {coin, coinlist}];
ways[[amount]])
